metals/src/main/scala/scala/meta/internal/metals/Warnings.scala

```diff
@@ -76,7 +76,7 @@ final class Warnings(
     }
     isReported match {
       case Some(()) =>
-      case None =>
+      case None if !path.isSbt =>
```

Comment:
Otherwise we get warnings that are not neccessary here.

Reply:
I started thinking to suppress these warnings for worksheets too.
But they produce warnings at a different place here:
https://github.com/scalameta/metals/blob/21fa274e18a1547ef511c41f3e036e03a89e5a00/metals/src/main/scala/scala/meta/internal/metals/Warnings.scala#L70-L73

Maybe, it would be better to move `BuildTargets.sbtBuildTarget` logic into `BuildTargets.inverseSources` as you orinially proposed and move this check at line 70 for `.sbt` and `.worksheet.sc`?

Reply:
Sure, fixed both!